### DISCUSSION TITLE
Best Practices and Updated Samples

### DISCUSSION THREAD

**Discussion by zorgiepoo** [sparkle-project/Sparkle#2262] [Announcements]
I have updated our documentation by:

* Re-writing the [SwiftUI example](https://sparkle-project.org/documentation/programmatic-setup#create-an-updater-in-swiftui) with better logical structure
* Adding an example of [setting up the updater with Qt](https://sparkle-project.org/documentation/programmatic-setup/#create-an-updater-in-qt) and useful information for integrating Sparkle outside of Xcode
* Adding examples of [adding updater settings with both SwiftUI and Qt](https://sparkle-project.org/documentation/preferences-ui/#adding-settings-in-swiftui)
* Highlight the [Core APIs and expectations of non-core APIs](https://sparkle-project.org/documentation/programmatic-setup/#core-apis) (i.e, what not to do) that are easy to get 'wrong' if the API documentation is not carefully read. For example, updater settings like `automaticallyChecksForUpdates` should only be set upon user preference changes and don't need to be backed by extra defaults storage.

If anything feels off, feel free to add to the discussion here or [help improve the website](https://github.com/sparkle-project/sparkle-project.github.io).

**Comment by patratacus** [comment]
How do you change the "Info.plist" now that it doesn't exist anymore in Xcode 14?  I found a bunch of hacks to bring back Info.plist but they don't seem reliable.  Somebody pointed out that the information that is in the Info.plist is now under the Target -> Info -> Custom macOS Application Target Properties

I don't see "SUFeedURL" or "CFBundleVersion" there but I think I can just add it using the "+" when highlighting any of the existing keys.  I haven't tried this yet since I just discovered your SwiftUI example just now.  I don't know whether you already know how to already set this up with Xcode 14.

**Reply by zorgiepoo** [reply]
In the `Info` tab of your target (besides the `General`, `Signing & Capabilities`, `Resource Tags` tabs) , you can add a new entry and set its `Key` to be `SUFeedURL`. An `Info.plist` file will then also show up in the project navigator, which may only contain custom user-defined keys like this one (and therefore is a subset of all the keys).

Anyway, this is a Xcode UI thing. At the end of the day, an Info.plist file will be generated inside your app bundle still. I think this change was done as part of Xcode 13 too.
